packages/near-membrane-base/src/membrane.ts

```diff
@@ -880,10 +896,8 @@ export function createMembraneMarshall() {
 
             private makeProxyUnambiguous(shadowTarget: ShadowTarget) {
                 if (
-                    foreignCallableHasOwnProperty(
-                        this.foreignTargetPointer,
-                        LOCKER_LIVE_MARKER_SYMBOL
-                    )
+                    this.targetTraits & TargetTraits.IsArray ||
```

Comment:
The reason why this is needed is because anything passing the test of isArray should be considered live, subclassing arrays is not very common, and even if that's the case, the length behavior and such seems like a pretty good indication that should be considered live.